### PR TITLE
Fix  road attributes

### PR DESCRIPTION
**Pull request by alisa911**

### CODE REVIEW

**Comment by alisa911** [comment]
https://github.com/osmandapp/OsmAnd-Issues/issues/2088

**Comment by alisa911** [comment]
Screenshot 2023-09-25 at 13 19 40
Screenshot 2023-09-25 at 13 19 48
Screenshot 2023-09-25 at 13 20 05

File: map/src/infoblock/components/graph/GpxGraph.jsx [hunk 1]
```diff
@@ -391,10 +417,15 @@ export default function GpxGraph({
             x: {
                 display: true,
                 type: 'linear',
-                max: data[data.length - 1][xAxis],
+                max: Number(data[data.length - 1][xAxis].toFixed(1)),
```
Comment (RZR-UA) [inline map/src/infoblock/components/graph/GpxGraph.jsx:420]: Did you mean:

max: Number(data[data.length - 1][xAxis]).toFixed(1),

?

File: map/src/infoblock/components/graph/RoadAttributesGraph.jsx [hunk 2]
```diff
@@ -77,84 +99,111 @@ export default function RoadAttributesGraph({ name, data, width }) {
         return type;
     }
 
+    function prepareDistance(value) {
+        if (value?.distance) {
+            if (value.distance > 0.01) {
+                return `${(value.distance / 1000).toFixed(2)} km`;
+            } else {
+                return '< 10 m';
+            }
```
Comment (RZR-UA) [inline map/src/infoblock/components/graph/RoadAttributesGraph.jsx:108]: value.distance < 0.01 (_in meters?_) = '< 10m' ?

Possibly better to use value.distance < 10

File: map/src/infoblock/components/graph/RoadAttributesGraph.jsx [hunk 3]
```diff
@@ -77,84 +99,111 @@ export default function RoadAttributesGraph({ name, data, width }) {
...
+                </Grid>
+            </Grid>
+            <Box sx={{ p: 0, width: Number(width.replace('px', '')) - 42, height: 60 }}>
                 <Bar
                     margin={{ top: 0, right: 0, left: -20, bottom: 0 }}
                     style={{ fontSize: 10 }}
                     data={graphData}
                     options={options}
                     onMouseMove={(e) => onMouseMoveGraph(e, chartRef)}
+                    onMouseLeave={() => hideSelectedPointSegment()}
```
Comment (RZR-UA) [inline map/src/infoblock/components/graph/RoadAttributesGraph.jsx:161]: onMouseLeave={hideSelectedPointSegment}

No need to create new function () => xxx() every time when you not use arguments such as: (e) => xxx(e)

**Review by RZR-UA — APPROVED** [review]

File: map/src/infoblock/components/graph/RoadAttributesGraph.jsx [hunk 4]
```diff
@@ -89,6 +89,20 @@ export default function RoadAttributesGraph({ name, data, width, selectedPoint }
         datasets: data.datasets,
     };
 
+    useEffect(() => {
+        if (activeIndex) {
+            if (activeIndex) {
```
Comment (RZR-UA) [inline map/src/infoblock/components/graph/RoadAttributesGraph.jsx:94]: Double if ?

**Review by RZR-UA — CHANGES_REQUESTED** [review]

**Review by RZR-UA — APPROVED** [review]

**Pull request merged**